### PR TITLE
added_Order_model-firstCommit

### PR DESCRIPTION
**Pull request by OlegShvets1**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/service/OrderService.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.List;
+import mate.academy.bookstore.dto.order.OrderItemResponseDto;
+import mate.academy.bookstore.dto.order.OrderPresentationRequestDto;
+import mate.academy.bookstore.dto.order.OrderResponseDto;
+import mate.academy.bookstore.dto.order.OrderStatusRequestDto;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+
+public interface OrderService {
+    OrderResponseDto createOrder(Authentication authentication, Pageable pageable,
```
Comment (Vasul-Pankiv) [inline src/main/java/mate/academy/bookstore/service/OrderService.java]: better to pass just userId here not Authentication (same with other methods)

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        User user = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId()).orElseThrow(
+                () -> new EntityNotFoundException("Can't find ShoppingCart by userId: "
+                        + user.getId()));
+
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        if (cartItems.isEmpty()) {
+            throw new OrderException("ShoppingCart is empty. "
+                    + "Please add items to it before confirming your order.");
+        }
```
Comment (Vasul-Pankiv) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java]: I think better just return empty list, it is general behavior for getAll() methods

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                .collect(Collectors.toSet());
+        orderItemRepository.saveAll(orderItems);
+        return orderItems;
+    }
+
+    private OrderItem fullFillOrderItem(Order order, CartItem cartItem) {
+        OrderItem orderItem = orderItemMapper.toModelFromCartItem(cartItem);
+        orderItem.setOrder(order);
+        return orderItem;
+    }
```
Comment (Vasul-Pankiv) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java]: bad naming, you are not fulfilling, but just creating order items.

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+    private Order getValidOrder(Authentication authentication, Long orderId) {
```
Comment (Vasul-Pankiv) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java]: Bad naming. The name should represent what the method does. But what is a valid Order?

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        Order order = placeAnOrder(shoppingCart, requestDto.getShippingAddress());
+        order.setOrderItems(fullFillSetOfOrderItems(order, cartItems));
+        OrderResponseDto orderResponseDto = orderMapper.toDto(order);
+
+        cleanShoppingCart(shoppingCart);
+        return orderResponseDto;
+    }
+
+    @Override
+    @Transactional
```
Comment (Vasul-Pankiv) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java]: why it is transactional, it is just get

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    public List<OrderResponseDto> getOrders(Authentication authentication, Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+        return orderRepository.findOrdersByUserId(user.getId(), pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
```
Comment (Vasul-Pankiv) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java:70]: Update method has to be transactional

File: src/main/resources/db/changelog/changes/09-create-orders-table.yaml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                  name: order_date
+                  type: timestamp
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (Vasul-Pankiv) [inline src/main/resources/db/changelog/changes/09-create-orders-table.yaml]: add empty line to the end of file

File: src/main/resources/db/changelog/changes/10-create-order-items-table.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                  name: price
+                  type: decimal
+                  constraints:
+                    nullable: false
+              - column:
+                 name: is_deleted
+                 type: BOOLEAN
+                 constraints:
+                  nullable: false
+                  defaultValueBoolean: false
```
Comment (Vasul-Pankiv) [inline src/main/resources/db/changelog/changes/10-create-order-items-table.yaml]: add empty line to the end of file

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 9]
```diff
@@ -14,4 +14,8 @@ databaseChangeLog:
   - include:
       file: db/changelog/changes/07-create-shopping-carts-table.yaml
   - include:
-      file: db/changelog/changes/08-create-cart-items-table.yaml
\ No newline at end of file
+      file: db/changelog/changes/08-create-cart-items-table.yaml
+  - include:
+      file: db/changelog/changes/09-create-orders-table.yaml
+  - include:
+      file: db/changelog/changes/10-create-order-items-table.yaml
```
Comment (Vasul-Pankiv) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: add empty line to the end of file

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request merged**